src/python/pants/backend/adhoc/run_system_binary.py

```diff
@@ -58,13 +59,11 @@ class SystemBinaryFieldSet(RunFieldSet):
 async def _find_binary(
     address: Address,
     binary_name: str,
-    extra_search_paths: Iterable[str],
+    search_paths: SearchPath,
```

Comment:
This should probably be named `search_path`, for consistency with the type name. It is confusing, because the PATH (singular) is actually a collection of multiple directory paths. But since the type name is singular, I guess it means "Path" in the sense of the PATH env var, so we should go for that consistently. 

If it were a list of strings, than using the plural would make sense.